internal/cli/cli.go

```diff
@@ -38,83 +42,180 @@ type CommandDefinition struct {
 }
 
 // CommandFactory is a command factory
-type CommandFactory struct {
-	Profile *Profile
-	Config  *Config
+type CommandFactory interface {
+	Build(func() CommandDefinition) *cobra.Command
+	Close()
+	Init()
+	Run(cmd *cobra.Command)
+	SetGlobalFlags(fs *flag.FlagSet)
+}
+
+type commandFactory struct {
+	in      *os.File
+	out     *os.File
+	err     *os.File
+	config  *Config
+	profile *Profile
+	ui      terminal.UI
+	logger  *log.Logger
 }
 
 // Config is the global CLI config
 type Config struct {
-	Command CommandConfig
-	UI      terminal.UIConfig
+	CommandConfig
+	terminal.UIConfig
 }
 
 // CommandConfig holds the global config for a CLI command
 type CommandConfig struct {
 	RealmBaseURL string
 }
 
+// NewCommandFactory creates a new command factory
+func NewCommandFactory() CommandFactory {
+	logger := log.New(os.Stderr, "UTC ERROR ", log.Ltime|log.Lmsgprefix)
```

Comment:
logger is always Stderr?

Reply:
yeah, `logger` probably isn't the right word for this, but this logger is created/kept for the sole purposes of calling `logger.Fatal` when a `ui` is not yet available.

`errLogger` felt verbose at the time, but I'll admit it pairs nicely with `errWriter` now. I'll update the name here